### ISSUE TITLE
1313

### ISSUE THREAD

**Issue by Bahubali120** [Bahubali120/KA-49#4]
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
df=pd.read_csv('/content/auto_mpg.csv')
df

df.drop('name',axis=1,inplace=True)

from sklearn.preprocessing import LabelEncoder
le=LabelEncoder()
df['origin']=le.fit_transform(df['origin'])
df

df.isnull().sum()

df.fillna(method='ffill',inplace=True)
df.isnull().sum()

x=df.iloc[:,1:8]
x

y=df[['mpg']]
y

from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.2,random_state=42)

x_train
y_train
x_test
y_test

from sklearn.linear_model import LinearRegression
lr=LinearRegression()
lr.fit(x_train,y_train)

y_pred=lr.predict(x_test)
y_pred

from sklearn.metrics import mean_squared_error,mean_absolute_error,r2_score
mse=mean_squared_error(y_test,y_pred)
mae=mean_absolute_error(y_test,y_pred)
r2=r2_score(y_test,y_pred)
rmse=np.sqrt(mse)
print('Mean Squared Error:',mse)
print('Mean Absolute Error:',mae)
print('Root Mean Squared Error:',rmse)
print('R2 Score:',r2)
